src/main/java/net/smartcosmos/dao/relationships/impl/RelationshipPersistenceService.java

```diff
@@ -32,237 +39,284 @@
-        // TODO: Implement method
-        throw new UnsupportedOperationException("Not implemented yet");
-    }
-
-    // endregion
+    public Optional<RelationshipResponse> create(String tenantUrn, RelationshipCreate createRelationship) {
 
-    // region Delete
+        if (alreadyExists(tenantUrn, createRelationship)) {
+            return Optional.empty();
+        }
+        RelationshipEntity entity = null;
+        try {
+            UUID tenantId = UuidUtil.getUuidFromUrn(tenantUrn);
+            entity = conversionService.convert(createRelationship, RelationshipEntity.class);
+            entity.setTenantId(tenantId);
+            entity = persist(entity);
+
+        } catch (Exception e) {
+            exceptionLogger("RelationshipPersistenceService:create", e, tenantUrn, createRelationship);
+            throw e;
+        }
+        return Optional.ofNullable(conversionService.convert(entity, RelationshipResponse.class));
+    }
+    //endregion
 
+    // region delete
     @Override
     public List<RelationshipResponse> delete(String tenantUrn, String urn) {
 
-        UUID accountId = UuidUtil.getUuidFromUrn(tenantUrn);
+        UUID tenantId = UuidUtil.getUuidFromUrn(tenantUrn);
 
-        List<RelationshipEntity> deleteList = new ArrayList<>();
         try {
             UUID uuid = UuidUtil.getUuidFromUrn(urn);
-            deleteList = relationshipRepository.deleteByTenantIdAndId(accountId, uuid);
-        } catch (IllegalArgumentException e) {
-            // empty list will be returned anyway
-            log.warn("Illegal URN submitted: %s by account %s", urn, tenantUrn);
-        }
+            return convertList(relationshipRepository.deleteByTenantIdAndId(tenantId, uuid));
 
-        return convert(deleteList);
+        } catch (Exception e) {
+            exceptionLogger("RelationshipPersistenceService:delete", e, "tenantUrn: " + tenantUrn, "urn: " + urn);
+        }
+        return new ArrayList<>();
     }
-
     // endregion
 
     // region Find specific / by URN
-
     @Override
     public Optional<RelationshipResponse> findByUrn(String tenantUrn, String urn) {
 
-        UUID accountId = UuidUtil.getUuidFromUrn(tenantUrn);
-
+        UUID tenantId = UuidUtil.getUuidFromUrn(tenantUrn);
         Optional<RelationshipEntity> entity = Optional.empty();
+
         try {
             UUID uuid = UuidUtil.getUuidFromUrn(urn);
-            entity = relationshipRepository.findByTenantIdAndId(accountId, uuid);
-        }
-        catch (IllegalArgumentException e) {
-            // Optional.empty() will be returned anyway
-            log.warn("Illegal URN submitted: %s by account %s", urn, tenantUrn);
+            entity = relationshipRepository.findByTenantIdAndId(tenantId, uuid);
+        } catch (IllegalArgumentException e) {
+            exceptionLogger("RelationshipPersistenceService:findByUrn", e, "tenantUrn: " + tenantUrn, "urn: " + urn);
```

Comment:
This is not helpful.  See my example in the other comment.  The classname is already written by the logger as long as we have it configured properly and the pertinent information is being placed into parms.  Just build out the string and log it here.  This is not good reuse